File: MPP/Laborator1/src/test/java/repository/UserRepositoryTest.java
package repository;

import model.TipUser;
import model.User;
import org.junit.Test;

import java.io.FileReader;
import java.io.IOException;
import java.util.List;
import java.util.Properties;
import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

import static org.junit.Assert.*;

public class UserRepositoryTest {
    @Test
    public void test(){
        Properties prop=new Properties();

        try {
            prop.load(new FileReader("D:\\University-Courses\\MPP\\Laborator1\\src\\test\\resources\\bd.properties"));
        } catch (IOException e) {
            e.printStackTrace();
        }

        IRepositoryUser repo=new UserRepository(prop);

        User user=new User("ioana_l","hashhash");

        //size
        assertEquals(3,repo.size());

        //save
        repo.save(user);
        assertEquals(4,repo.size());

        //find
        assertNull(repo.findOne("teofana"));
        assertEquals(user,repo.findOne("ioana_l"));

        //update
        user.setTip(TipUser.OPERATOR);
        repo.update("ioana_l",new User("ioana_l","hashhash",TipUser.OPERATOR));

        List<User> lista= StreamSupport.stream(repo.findAll().spliterator(),false)
                .collect(Collectors.toList());
        User lastUser=lista.get(lista.size()-1);
        assertEquals(user,lastUser);

        //delete
        repo.delete("ioana_l");
        assertEquals(3,repo.size());

    }


}

File: MPP/LabC#/UnitTest/UserRepositoryTest.cs
using System;
using System.Collections.Generic;
using Concurs.model;
using log4net.Config;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UnitTest
{
    [TestClass]
    public class UserRepositoryTest
    {
        [TestMethod]
        public void TestMethod1()
        {
            /*XmlConfigurator.Configure(new System.IO.FileInfo("App.config"));

            IDictionary<string, string> props = new SortedList<String, String>();

            props.Add("ConnectionString", Concurs.repository.DBUtils.GetConnectionStringByName("catalogTest"));

            UserRepository repo = new UserRepository(props);

            User user = new User("ioana_l", "hashhash");

            //size
            Assert.AreEqual(3, repo.Size());

            //save
            repo.Save(user);
            Assert.AreEqual(4, repo.Size());

            //find
            Assert.IsNull(repo.FindOne("teofana"));
            Assert.AreEqual(user, repo.FindOne("ioana_l"));

            //update
            user.Tip="OPERATOR";
            repo.Update("ioana_l", new User("ioana_l", "hashhash", "OPERATOR"));

            List<User> lista = (List<User>)repo.FindAll();
            User lastUser = lista[lista.Count - 1];
            Assert.AreEqual(user, lastUser);

            try
            {
                repo.Update("ioa",user);
                Assert.Fail();
            }
            catch (RepositoryException e)
            {
                Assert.AreEqual(e.Message, "Error: Nu s-a putut actualiza userul!");
            }

            //delete
            repo.Delete("ioana_l");
            Assert.AreEqual(3, repo.Size());
            try
            {
                repo.Delete("ioana_l");
                Assert.Fail();
            }
            catch(RepositoryException e)
            {
                Assert.AreEqual(e.Message, "Error: Userul nu s-a putut sterge!");
            }*/

        }

    }
}

